File: src/MetadataService/Models/Tree/FolderNode.cs
namespace MetadataService.Models.Tree;

public class FolderNode
{
            public string Name { get; set; }
            public List<FolderNode> Folders { get; set; } = new();
            public List<FileNode> Files { get; set; } = new();

            public DateTime? UploadedAt { get; set; } // new
            public string? UploadedByName { get; set; } // new
    

    public FolderNode(string name)
    {
        Name = name;
    }
}

File: frontend/src/app/models/folder-node.ts
export interface FolderNode {
  name: string;
  folders: FolderNode[];
  files: {
    id: string;
    name: string;
    uploadedAt?: string;
    uploadedByName?: string;
    contentType?: string;
    size?: number; // only files have size
  }[];
  uploadedAt?: string;
  uploadedByName?: string;
}
